substrate/frame/staking-async/ahm-test/src/ah/test.rs

```diff
@@ -185,12 +136,71 @@ fn on_receive_session_report() {
 			]
 		);
 
+		// roll three more sessions...
+		for i in 1..=3 {
+			// roll some random number of blocks.
+			roll_many(10);
+
+			// send the session report.
+			assert_ok!(rc_client::Pallet::<T>::relay_session_report(
+				RuntimeOrigin::root(),
+				rc_client::SessionReport {
+					end_index: i,
+					validator_points: vec![(1, 10)],
+					activation_timestamp: None,
+					leftover: false,
+				}
+			));
+
+			let era_points = staking_async::ErasRewardPoints::<T>::get(&0);
+			assert_eq!(era_points.total, 360 + i * 10);
+			assert_eq!(era_points.individual.get(&1), Some(&(10 + i * 10)));
+
+			assert_eq!(
+				staking_events_since_last_call(),
+				vec![StakingEvent::SessionRotated {
+					starting_session: i + 1,
+					active_era: 0,
+					planned_era: 1
+				}]
+			);
+		}
+
+		// no election events emitted anymore
+		assert_eq!(election_events_since_last_call(), vec![]);
+
+		// Next session should export the validator set
+		assert_ok!(rc_client::Pallet::<T>::relay_session_report(
+			RuntimeOrigin::root(),
+			rc_client::SessionReport {
+				end_index: 4,
+				validator_points: vec![(1, 10)],
+				activation_timestamp: None,
+				leftover: false,
+			}
+		));
+
+		assert_eq!(
+			staking_events_since_last_call(),
+			vec![StakingEvent::SessionRotated {
+				starting_session: 5,
+				active_era: 0,
+				planned_era: 1
+			}]
+		);
+
+		// no xcm message sent yet.
+		assert_eq!(LocalQueue::get().unwrap(), vec![]);
+
+		// next block triggers to export
+		roll_next();
+
 		// New validator set xcm message is sent to RC.
 		assert_eq!(
```

Comment:
Can you extend this test to also show the immediate start of the next election? Or a new test for this that is more fine-grained than `era_lifecycle_test` and is using the defaults.